### PR TITLE
Add getters for immutables for LsETH and rETH

### PR DESCRIPTION
**Pull request by cashd**
- Added getter functions for immutables in LsETH/rETH integrations. 
- Added test for getter functions for code coverage.

### CODE REVIEW

File: contracts/src/interfaces/IRethHyperdriveRead.sol [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.20;
+
+import { IHyperdriveRead } from "./IHyperdriveRead.sol";
+import { IRocketStorage } from "./IRocketStorage.sol";
+import { IRocketTokenRETH } from "./IRocketTokenRETH.sol";
+
+interface IRethHyperdriveRead is IHyperdriveRead {
```
Comment (jalextowle) [inline contracts/src/interfaces/IRethHyperdriveRead.sol:8]: We should add a `IRethHyperdrive` interface that consumes this if we don't have one already. Also, looking at this again, I'm wondering if `RETH` would be more consistent in these names than `Reth`. We should consider changing this everywhere.

File: contracts/src/interfaces/lseth/ILsETHHyperdriveRead.sol [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.20;
+
+import { IHyperdriveRead } from "../IHyperdriveRead.sol";
+import { IRiverV1 } from "./IRiverV1.sol";
+
+interface ILsETHHyperdriveRead is IHyperdriveRead {
```
Comment (jalextowle) [inline contracts/src/interfaces/lseth/ILsETHHyperdriveRead.sol:7]: Same here, we should add a `ILsETHHyperdrive` interface that consumes this. Check out `IStETHHyperdrive` for an example.

File: contracts/src/interfaces/IRethHyperdriveRead.sol [hunk 3]
```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.20;
+
+import { IHyperdriveRead } from "./IHyperdriveRead.sol";
+import { IRocketStorage } from "./IRocketStorage.sol";
+import { IRocketTokenRETH } from "./IRocketTokenRETH.sol";
+
+interface IRETHHyperdriveRead is IHyperdriveRead {
```
Comment (jalextowle) [inline contracts/src/interfaces/IRethHyperdriveRead.sol:8]: This name looks right, but the filename is still `IRethHyperdriveRead`. The MacOS filesystem is case insensitive, so you might need to use `git mv` to actually update the file name.

**Review by jalextowle — APPROVED** [review]
LGTM once the remaining file is renamed.

File: contracts/src/interfaces/IRethHyperdriveRead.sol (hunk 3 above)
Comment (cashd) [reply contracts/src/interfaces/IRethHyperdriveRead.sol]: good catch, I had it updated but git did not register!!

**Pull request merged**